src/script-runner.js

```diff
@@ -34,26 +91,47 @@ window.web3Provider = {
 }
 window.web3 = new Web3(window.web3Provider)
 
-window.starknet = starknet
-
-console.log = function () {
-  window.remix.emit('log', {
-    data: Array.from(arguments)
+// 'getContractFactory' is added to 'ethers' to support usage of hardhat.ethers in the tests
+window.ethers.getContractFactory = (contractName, options) => {
+  let artifactsPath = options && options.artifactsPath || 'browser/contracts/artifacts'
```

Comment:
you shouldn't be using the artifacts file for that, but the `compilerArtefacts` plugin (see in the remix-core-plugin module).
what you can do for resolving the contract here:
 - first call `getLastCompilationResult` and look for the contract name in the list
 - if the name is not present, call `getAllContractDatas` and do the same.

Alternatively, you can add a new function in `compilerArtefacts` like  `getArtefactsByContractName`which will do the same, and use it from the script runner

Reply:
This is done.

Reply:
> you can add a new function in compilerArtefacts like getArtefactsByContractNam

you have to implement that function.

Reply:
Implemented here: https://github.com/ethereum/remix-project/pull/2073/commits/b2736c8e6bea005b926728f2c9b6d726f6095576